packages/ws-server/src/webSocketServer.ts

```
@@ -153,16 +170,24 @@ app.ws.use(async (ctx: Koa.Context) => {
         logger.trace(`${requestDetails.formattedLogPrefix}: Receive single request=${JSON.stringify(request)}`);
       }
 
+      if (request.method === 'eth_subscribe' && !areSubscriptionsEnabled()) {
+        const wsSubscriptionsDisabledError = predefined.WS_SUBSCRIPTIONS_DISABLED;
+        logger.warn(`${requestDetails.formattedLogPrefix}: ${JSON.stringify(wsSubscriptionsDisabledError)}`);
+        ctx.websocket.send(JSON.stringify(jsonResp(null, wsSubscriptionsDisabledError, undefined)));
+        return;
+      }
+
```

Hmmm, I’m not entirely sure if I’m making sense here, but this feels a bit out of place. This is like a top-level handler where we receive requests from the client, and it immediately checks for `eth_subscribe` and whether `SUBSCRIPTIONS_ENABLED` is true, which seems specific to the subscription service. Also, wouldn’t we want to throw the error when it’s the `eth_unsubscribe` method as well? 

It might make more sense to move this logic into `getRequestResult()`, within the switch statement where subscription-related cases are handled, or even within the service methods themselves, rather than leaving it in this global place.

---

I was looking at the check for batch requests, which is also in the webSocketServe. Now thinking about it, its best for the check to be in subscriptions controller. We also need to think  about the other checks in the if before

---

Yeah makes sense thanks!